app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt

```diff
@@ -296,6 +299,12 @@ class LocationSensorManager : BroadcastReceiver(), SensorManager {
         )
     }
 
+    override fun hasSensor(context: Context, sensorId: String): Boolean {
+        return if (sensorId == singleAccurateLocation.id && !DisabledLocationHandler.hasGPS(context)) {
```

Comment:
So is the goal to only hide one sensor here? Because to do that you just need to modify <https://github.com/home-assistant/android/blob/5a2e0476951ddb4b330661e7aeb5265f0ccf4215/app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt#L632> to return the sensors to be shown.

Reply:
Yes i want to remove the sensor. 

Other sensors does also use the `hasSensor` method to check if a sensor is available. The only problem was, that the `LocationSensoManager` has multiple sensors. That's why, i included the `sensorId` parameter to `hasSensor`.

If i change the logic to the `availableSensors` property, i would do it differently on the `LocationSensorManager` as on the other sensors. I like booth solutions. What solution do you prefer?

Reply:
`availableSensors` is designed to show which sensors are available meaning there will always be one so the manager is shown all the time.  `hasSensor` was created to hide the entire manager completely so it ignores `availableSensors` like in the case of LightSensorManager as it only has 1 sensor we want to hide the entire block.  So one hides the parent the other only the children.

I personally like to keep it separate because sometimes a device will support some sensors but not all like if they are on API24 and the manager has sensors in API23 and API 26 we can hide 26 with just the list and keep the manager shown with the others.

Reply:
Thanks for clarification. Now i understood the concept behind `hasSensor`. Then i agree fully with you 💯